pkg/cmd/pr/checks/aggregate.go

```diff
@@ -28,6 +29,10 @@ type checkCounts struct {
 	Canceled int
 }
 
+func (ch *check) ExportData(fields []string) map[string]interface{} {
+	return cmdutil.StructExportData(ch, fields)
+}
+
```

Comment:
**Is there anything about how PR status check roll ups are exported that should be considered here?**

https://github.com/cli/cli/blob/178fc2e51ead22bd0ef2904b4b5e3018d657914d/api/export_pr.go#L48-L76

```shell
$ gh pr view 9079 --json statusCheckRollup
{
  "statusCheckRollup": [
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:06Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558491/job/24952304248",
      "name": "CodeQL-Build",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Code Scanning"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:52:03Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558501/job/24952304198",
      "name": "lint",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Lint"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:52:04Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557717/job/24952302132",
      "name": "lint",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Lint"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:53Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952304215",
      "name": "build (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:53Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302733",
      "name": "build (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:35Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "0001-01-01T00:00:00Z",
      "conclusion": "",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305470",
      "name": "build (windows-latest)",
      "startedAt": "2024-05-14T13:50:40Z",
      "status": "IN_PROGRESS",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "0001-01-01T00:00:00Z",
      "conclusion": "",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303574",
      "name": "build (windows-latest)",
      "startedAt": "2024-05-14T13:50:43Z",
      "status": "IN_PROGRESS",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:26Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306154",
      "name": "build (macos-latest)",
      "startedAt": "2024-05-14T13:50:35Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:20Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304747",
      "name": "build (macos-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:50:49Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305094",
      "name": "integration-tests (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:50:49Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302126",
      "name": "integration-tests (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:51Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305791",
      "name": "integration-tests (windows-latest)",
      "startedAt": "2024-05-14T13:50:40Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:00Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303150",
      "name": "integration-tests (windows-latest)",
      "startedAt": "2024-05-14T13:50:39Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:51:05Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306752",
      "name": "integration-tests (macos-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:51:03Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304015",
      "name": "integration-tests (macos-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:00Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/runs/24952426191",
      "name": "CodeQL",
      "startedAt": "2024-05-14T13:52:57Z",
      "status": "COMPLETED",
      "workflowName": ""
    }
  ]
}
```

versus

```shell
$ ./bin/gh pr checks 9079 --json bucket,completedAt,description,event,link,name,startedAt,state,workflow
[
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:00Z",
    "description": "",
    "event": "",
    "link": "https://github.com/cli/cli/runs/24952426191",
    "name": "CodeQL",
    "startedAt": "2024-05-14T13:52:57Z",
    "state": "SUCCESS",
    "workflow": ""
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:59:59Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303574",
    "name": "build (windows-latest)",
    "startedAt": "2024-05-14T13:50:43Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pending",
    "completedAt": "0001-01-01T00:00:00Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305470",
    "name": "build (windows-latest)",
    "startedAt": "2024-05-14T13:50:40Z",
    "state": "IN_PROGRESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:51Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305791",
    "name": "integration-tests (windows-latest)",
    "startedAt": "2024-05-14T13:50:40Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:00Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303150",
    "name": "integration-tests (windows-latest)",
    "startedAt": "2024-05-14T13:50:39Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:50:49Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305094",
    "name": "integration-tests (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:51:05Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306752",
    "name": "integration-tests (macos-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:53Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952304215",
    "name": "build (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:06Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558491/job/24952304248",
    "name": "CodeQL-Build",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Code Scanning"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:20Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304747",
    "name": "build (macos-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:50:49Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302126",
    "name": "integration-tests (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:52:04Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557717/job/24952302132",
    "name": "lint",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Lint"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:51:03Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304015",
    "name": "integration-tests (macos-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:52:03Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558501/job/24952304198",
    "name": "lint",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Lint"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:26Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306154",
    "name": "build (macos-latest)",
    "startedAt": "2024-05-14T13:50:35Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:53Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302733",
    "name": "build (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:35Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  }
]
```

Reply:
Unless there's a specific reason, I think having parity between the two outputs would be better. It seems that the `statusCheckRollup` is missing `bucket`, `description`, and `event` and has some different names 🤔 

`gh pr checks --json` | `gh pr view --json statusCheckRollup`
---                   | ---
bucket                |
completedAt           | completedAt
description           |
event                 |
link                  | detailsUrl
name                  | name
startedAt             | startedAt
state                 | conclusion
workflow              |
_                     | status

Reply:
Agree being consistent is valuable.

Reply:
That's fair.

Reply:
created https://github.com/cli/cli/issues/9091 to follow up